app/workers/s3_raw.rb

```diff
@@ -19,27 +20,25 @@ def work
       write_location = "s3://#{bucket.name}/#{key}"
       logger.info("Starting batch from #{queue.url} to #{write_location}")
 
-      batched_messages = []
-      loop do
+      count = 0
+      body = StringIO.new
+      1_000.times do # Getting 10_000 (1_000 * 10) messages at most.
         messages = queue.receive_messages(max_number_of_messages: 10)
-        batched_messages += messages.to_a
-        break if messages.size == 0
+        break if messages.size.zero?
+        count += messages.size
+        messages.each do |msg|
```

Comment:
Instead of saving all `messages` in memory, it receives messages and processes them immediately.